pkg/storegateway/series_refs.go

```diff
@@ -830,69 +832,80 @@ func metasToChunks(blockID ulid.ULID, metas []chunks.Meta) []seriesChunkRef {
 	return chks
 }
 
-type seriesForPostingsCacheEntry struct {
-	LabelSets   []labels.Labels
-	MatchersKey indexcache.LabelMatchersKey
-	Shard       sharding.ShardSelector
-}
-
 func fetchCachedSeriesForPostings(ctx context.Context, userID string, indexCache indexcache.IndexCache, blockID ulid.ULID, matchers []*labels.Matcher, shard *sharding.ShardSelector, postingsKey indexcache.PostingsKey, logger log.Logger) (seriesChunkRefsSet, bool) {
 	matchersKey := indexcache.CanonicalLabelMatchersKey(matchers)
 	data, ok := indexCache.FetchSeriesForPostings(ctx, userID, blockID, matchersKey, shard, postingsKey)
 	if !ok {
 		return seriesChunkRefsSet{}, false
 	}
-
-	var entry seriesForPostingsCacheEntry
-	if err := decodeSnappyGob(data, &entry); err != nil {
+	data, err := snappy.Decode(nil, data)
```

Comment:
I think it's worth pooling the buffer provided to `snappy.Decode` here, wdyt?

Reply:
yeah i can agree. It's accounting for 10% of allocations of the store-gateway. 

I'm reluctant to do it since I'm not sure about the effectiveness of those and whether we won't end up with pooled slices much larger than what we actually need - I assume some things will be mere hundreds of bytes and other will be megabytes. Maybe we can use the pool.BucketBytes. Do you have suggestions for the sizes of buckets?